frontend/src/components/common/Notifications/index.tsx

```diff
@@ -0,0 +1,210 @@
+import bellIcon from '@iconify/icons-mdi/bell';
+import { Icon } from '@iconify/react';
+import {
+  Badge,
+  Grid,
+  IconButton,
+  Link,
+  makeStyles,
+  Theme,
+  Tooltip,
+  Typography,
+} from '@material-ui/core';
+import { Menu, MenuItem } from '@material-ui/core';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { Link as RouterLink } from 'react-router-dom';
+import Event, { KubeEvent } from '../../../lib/k8s/event';
+import { Notification } from '../../../lib/notification';
+import { createRouteURL } from '../../../lib/router';
+import { timeAgo } from '../../../lib/util';
+import { setUINotifications, updateUINotification } from '../../../redux/actions/actions';
+import { useTypedSelector } from '../../../redux/reducers/reducers';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  notificationItem: {
+    '&.MuiMenuItem-root': {
+      borderBottom: `1px solid ${theme.palette.notificationBorderColor}`,
+      padding: '1rem',
+    },
+  },
+  notificationMessage: {
+    '&': {
+      wordBreak: 'break-word',
+      whiteSpace: 'normal',
+    },
+  },
+  root: {
+    '& .MuiPaper-root': {
+      width: '30vw',
+      minWidth: '300px',
+      maxHeight: '400px',
+    },
+  },
+  errorItem: {
+    '&': {
+      color: theme.palette.error,
+    },
+  },
+}));
+
+function prepareNotificationsList(
+  notifications: Notification[],
+  clickEventHandler: (notification?: Notification) => void
+) {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  if (!notifications || notifications.length === 0) {
+    return (
+      <MenuItem onClick={() => clickEventHandler()} className={classes.notificationItem}>
+        {t(`frequent|You don't have any notifications right now`)}
+      </MenuItem>
+    );
+  }
+
+  function menuItemClickHandler(notification?: Notification) {
+    if (!notification) {
+      return;
+    }
+    notification.seen = true;
+    clickEventHandler(notification);
+  }
+
+  function prepareNotificationLinkComponent(notification: Notification) {
+    if (!notification.url) {
+      return notification.message;
+    }
+    return (
+      <Link component={RouterLink} to={notification.url}>
+        {notification.message}
+      </Link>
+    );
+  }
+
+  return notifications?.map((notification, index: number) => {
+    return (
+      <MenuItem
+        onClick={() => menuItemClickHandler(notification)}
+        key={`${notification}__${index}`}
+        className={classes.notificationItem}
+      >
+        <Grid container spacing={4}>
+          <Grid item direction="column" md={notification.seen ? 12 : 11}>
+            <Grid item md={12}>
+              <Typography style={{ fontWeight: notification.seen ? 'normal' : 'bold' }}>
+                {prepareNotificationLinkComponent(notification)}
+              </Typography>
+            </Grid>
+            <Grid item>{timeAgo(notification.date)} ago</Grid>
```

Comment:
This is not translatable. I think `timeAgo` can already give the full sentence with "ago" in it, but you'd have to check if that's so.